### ISSUE TITLE
Popover positioned partially off-screen and cropped

### ISSUE THREAD

**Issue by peterblazejewicz** [ng-bootstrap/ng-bootstrap#2924]
The `top` popover, when triggered near the page border, is being rendered in such way it gets cropped:
https://stackblitz.com/angular/rdokjdpkdoq
https://ng-bootstrap.github.io/#/components/popover/examples

![image](https://user-images.githubusercontent.com/14539/49836572-750bbc00-fda3-11e8-81cf-d5d794944a18.png)
                                                                              
The BS uses popper.js, rendering in 4.1.3:

![image](https://user-images.githubusercontent.com/14539/49836690-dcc20700-fda3-11e8-93be-236daf984b04.png)

ng-bootstrap: v4.0.0
Chrome, macOS

**Comment by maxokorokov** [comment]
Ok, will reopen this as a feature request. 

In case of the `top` placement above, bootstrap/popper **SHIFT** the popover horizontally until it fits (note the arrow is not centered), we will just crop it if **ONLY** `top` is specified → we don't support this feature.

P.S. Workaround would be using `[position]="['top', 'auto']"`

**Comment by Sachin4dotnet** [comment]
Any update on this ?

**Comment by pelgro** [comment]
Pretty sure this is a bug. Secondary positioning should be applied to the arrow and not the whole popover window. At least that's how it works in popper.js

placement="top" results in top-center placement in positionElements. I believe the center placement should apply to the arrow and not the whole window. That way the popover will always be in the viewport as it's only moved above the reference object.

**Comment by ZiluckMichael** [comment]
@maxokorokov Any update? This is an incredibly annoying shortcoming of using popovers in this library.

**Comment by janwidmer** [comment]
Any Update on this bug?

**Comment by dmrickey** [comment]
Yeah `'auto'` has issues on its own and this is pretty annoying..
